### ISSUE TITLE
Issue in manipulation of resource output/reference with count

### ISSUE THREAD

**Issue by farman022** [hashicorp/terraform#19549]
Hi,
I have two resources. one with count variable and taking the reference of one in other. It will give error if the count is zero in the first one.

i.e 
I have one option group resource.

```
resource "aws_db_option_group" "rds_option_group" {
 count     = "${var.abc == "abc" ? 0 : 1}"
 name                 = "asdf-og"
 engine_name          = "sqlserver-se"
 major_engine_version = "xxxxx"

 option {
   option_name = "XXXXXXXX"

   option_settings {
     name  = "IAM_ROLE_ARN"
     value = "xxxxxx"
   }
 }
}
```

and in rds resource, I have written like that. Do not have count variable here in rds resource.

```
option_group_name  = "${var.bac == "abc" ? "some-value" : aws_db_option_group.rds_option_group.name}"
```

**Error i got if **var.abc != "abc"**

```
Error: Error running plan: 1 error(s) occurred:

* module.test.module.rds.aws_db_instance.rds: 1 error(s) occurred:

* module.test.module.rds.aws_db_instance.rds: Resource 'aws_db_option_group.rds_option_group' not found for variable 'aws_db_option_group.rds_option_group.name'**
```

Can you please let me know if we have any method to manipulate this. or it is a bug.

**Comment by farman022** [comment]
@apparentlymart can you look into it.

**Comment by mildwonkey** [comment]
Hi @farman022, sorry you have run across this.

I suspect you are running into a known limitation of the terraform's configuration language - both sides of a conditional are always evaluated, so you can get errors like this when a resource is not created, but referenced in a conditional.

Here is the main issue where we've been tracking this: 
https://github.com/hashicorp/terraform/issues/15605

This is (unfortunately)expected behavior in current versions of terraform, fixed in the upcoming 0.12 release.

**Comment by mildwonkey** [comment]
I am going to close this since it is fixed in 0.12. Please open a new issue if you have any other problems.
